### ISSUE TITLE
Cannot run `pnpm build` after importing `imageGenerationPlugin` to use in a character 

### ISSUE THREAD

**Issue by YoungPhlo** [elizaOS/eliza#917]
**Describe the bug**

Build failure in `@ai16z/eliza package` due to unresolved dependency `@ai16z/plugin-image-generation`. The build process fails when attempting to resolve the image generation plugin import in `defaultCharacter.ts`.

**To Reproduce**

- 1. Clone the eliza repository (v0.1.5-alpha.5)
- 2. Import and add the `imageGenerationPlugin` to Eliza in `defaultCharacter.ts`
```ts
import { Character, ModelProviderName, Clients } from "./types.ts";
import { imageGenerationPlugin } from "@ai16z/plugin-image-generation";

export const defaultCharacter: Character = {
    name: "Eliza",
    username: "eliza",
    plugins: [imageGenerationPlugin],
```
- 3. Run `pnpm build` in the project root
- 4. Build fails with error message indicating it cannot resolve "@ai16z/plugin-image-generation" in defaultCharacter.ts

**Expected behavior**

The build process should successfully compile all packages including @ai16z/eliza. The @ai16z/plugin-image-generation package should be properly resolved as it's listed in the workspace packages.

**Screenshots**

N/A

**Additional context**

- Project version: v0.1.5-alpha.
- Build tool: turbo 2.3.3
- Package manager: pnpm
- The package @ai16z/plugin-image-generation is listed in the workspace scope but fails to resolve during build
- Error occurs in the ESM build step using tsup
- File causing error: packages/core/src/defaultCharacter.ts

**Exact error:**
```sh
USER@MACHINE MINGW64 /c/ML/agents/testeliza/eliza-v0.1.5-alpha.5 ((fb24df4f...))
$ pnpm build

> eliza@ build C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5
> turbo run build --filter=!eliza-docs

turbo 2.3.3

• Packages in scope: @ai16z/adapter-postgres, @ai16z/adapter-sqlite, @ai16z/adapter-sqljs, @ai16z/adapternt-discord, @ai16z/client-farcaster, @ai16z/client-github, @ai16z/client-telegram, @ai16z/client-twitter, @ai16z/plugin-buttplug, @ai16z/plugin-coinbase, @ai16z/plugin-conflux, @ai16z/plugin-evm, @ai16z/plugin-ai16z/plugin-node, @ai16z/plugin-solana, @ai16z/plugin-starknet, @ai16z/plugin-tee, @ai16z/plugin-trustdbsapp, create-eliza-app, eliza-client
• Running build in 34 packages
• Remote caching disabled
@ai16z/eliza:build: cache miss, executing 7efb30d0891da17b
create-eliza-app:build: cache hit, replaying logs 86473ad74fdfa109
create-eliza-app:build: 
create-eliza-app:build: > create-eliza-app@0.1.5-alpha.5 build C:\ML\agents\testeliza\eliza-v0.1.5-alpha.
create-eliza-app:build: > unbuild
create-eliza-app:build: 
create-eliza-app:build: ℹ Automatically detected entries: src/index [esm] [cjs]
create-eliza-app:build: ℹ Building create-eliza-app
create-eliza-app:build: ℹ Cleaning dist directory: ./dist
create-eliza-app:build: ✔ Build succeeded for create-eliza-app
create-eliza-app:build:   dist/index.cjs (total size: 1.65 kB, chunk size: 1.65 kB)
create-eliza-app:build: 
create-eliza-app:build:   dist/index.mjs (total size: 1.64 kB, chunk size: 1.64 kB)
create-eliza-app:build: 
create-eliza-app:build: Σ Total dist size (byte size): 3.29 kB
create-eliza-app:build: 
@ai16z/eliza:build: 
@ai16z/eliza:build: > @ai16z/eliza@0.1.5-alpha.5 build C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\packag
@ai16z/eliza:build: > tsup --format esm --dts
@ai16z/eliza:build: 
@ai16z/eliza:build: CLI Building entry: src/index.ts
@ai16z/eliza:build: CLI Using tsconfig: tsconfig.json
@ai16z/eliza:build: CLI tsup v8.3.5
@ai16z/eliza:build: CLI Using tsup config: C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\packages\core\tsup
@ai16z/eliza:build: CLI Target: esnext
@ai16z/eliza:build: CLI Cleaning output folder
@ai16z/eliza:build: ESM Build start
@ai16z/eliza:build: X [ERROR] Could not resolve "@ai16z/plugin-image-generation"
@ai16z/eliza:build: 
@ai16z/eliza:build:     src/defaultCharacter.ts:2:38:
@ai16z/eliza:build:       2 │ ...rt { imageGenerationPlugin } from "@ai16z/plugin-image-generation";
@ai16z/eliza:build:         ╵                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@ai16z/eliza:build: 
@ai16z/eliza:build:   You can mark the path "@ai16z/plugin-image-generation" as external to exclude it frhe bundle.
@ai16z/eliza:build: 
@ai16z/eliza:build: ESM Build failed
@ai16z/eliza:build: Error: Build failed with 1 error:
@ai16z/eliza:build: src/defaultCharacter.ts:2:38: ERROR: Could not resolve "@ai16z/plugin-image-generation"
@ai16z/eliza:build:     at failureErrorWithLog (C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_modules\
@ai16z/eliza:build:     at C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_modules\esbuild\lib\main.js:9
@ai16z/eliza:build:     at runOnEndCallbacks (C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_modules\es
@ai16z/eliza:build:     at buildResponseToResult (C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_module
@ai16z/eliza:build:     at C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_modules\esbuild\lib\main.js:9
@ai16z/eliza:build:     at responseCallbacks.<computed> (C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node
@ai16z/eliza:build:     at handleIncomingPacket (C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_modules
@ai16z/eliza:build:     at Socket.readFromStdout (C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_module
@ai16z/eliza:build:     at Socket.emit (node:events:513:28)
@ai16z/eliza:build:     at addChunk (node:internal/streams/readable:559:12)
@ai16z/eliza:build: DTS Build start
@ai16z/eliza:build: src/defaultCharacter.ts(2,39): error TS2307: Cannot find module '@ai16z/plugin-image-
@ai16z/eliza:build: 
@ai16z/eliza:build: DTS Build error
za\eliza-v0.1.5-alpha.5\packages\core) C:\ML\agents\testeliza\eliza-v0.1.5-alpha.5\node_modules\.bin\pnpm.CMD run build exited (1)

 Tasks:    1 successful, 2 total
Cached:    1 cached, 2 total
  Time:    3.71s
Failed:    @ai16z/eliza#build

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by 0xMxxx** [comment]
found a solution? Im facing a similar issue

**Comment by CodingTux** [comment]
It will work if you add this plugin from here https://github.com/ai16z/eliza/blob/main/agent/src/index.ts, you can do so by pushing this inside "defaultCharacter" (https://github.com/ai16z/eliza/blob/main/agent/src/index.ts#L486) by adding `characters[0].plugins.push(imageGenerationPlugin);`. Provided code is only meant for testing.

**Comment by AIFlowML** [comment]
Hello everyone. 
did you solve this issue in hte latest version of ElizaOS ?
I post few steps to try fix but please let us know. 

### Comprehensive Solution

#### TLDR (Executive Summary)
- **Issue**: Build failure in `@ai16z/eliza` due to unresolved dependency `@ai16z/plugin-image-generation`.
- **Root Cause**: `@ai16z/plugin-image-generation` not correctly installed or linked in the workspace.
- **Solution**: Verify and reinstall dependencies, check import paths, and ensure correct build tool configuration.
- **Impact**: Disrupts development workflow and build process.

#### Quick Resolution Path
1. **Verify Installation**:
   - Check `node_modules` or `pnpm-lock.yaml` for `@ai16z/plugin-image-generation`.
   - Ensure it is listed in `package.json`:
     ```json
     "dependencies": {
       "@ai16z/plugin-image-generation": "workspace:*"
     }
     ```

2. **Reinstall Dependencies**:
   ```sh
   pnpm install
   ```

3. **Check Import Path**:
   - Verify the import statement in `packages/core/src/defaultCharacter.ts`:
     ```typescript
     import { imageGenerationPlugin } from '@ai16z/plugin-image-generation';
     ```

4. **Build the Plugin**:
   ```sh
   pnpm build
   ```

**Comment by jonathanmv** [comment]
Hey @YoungPhlo, 
try replacing the `ai16z` with `elizaos` as in `@elizaos/plugin-image-generation`.

**Comment by YoungPhlo** [comment]
> Hey @YoungPhlo, try replacing the `ai16z` with `elizaos` as in `@elizaos/plugin-image-generation`.

Just a heads up the `elizaOS` name change was on Dec 21 (https://github.com/elizaOS/eliza/commit/318a3a587cbf761607ba148d45e7c8ffe9e33b22) around 2 weeks after I opened this issue

**Comment by AIFlowML** [comment]
This issue need feedback and testing.

**Comment by rferrari** [comment]
I think this is happening with all plugins when you try to edit the file: packages/core/src/defaultCharacter.ts
I`m looking for how to load plugins creating a defaul Character and not using the characterer.json file. with json it can be loaded and works, but editing the core default is not.

> Cannot find module '@elizaos/plugin-solana' or its corresponding type declarations.ts(2307)
> Cannot find module '@elizaos/plugin-image-generation' or its corresponding type declarations.ts(2307)

import { Character, ModelProviderName } from "./types.ts";
import { solanaPlugin } from "@elizaos/plugin-solana";
//import { solanaPlugin } from "../../plugin-solana"; (also fail)
import { imageGenerationPlugin } from "@elizaos/plugin-image-generation";

export const defaultCharacter: Character = {
    name: "Eliza",
    username: "eliza",
    plugins: [solanaPlugin, imageGenerationPlugin],

**Comment by AIFlowML** [comment]
Hello here.
The .json loading is better because you not have to change the code in the core.

**Comment by rferrari** [comment]
yeah, but i want to learn how to do both ways... i already made some core changes to learn more about the code and play around

**Comment by brittanylcrocker** [comment]
I'm also experiencing the `Could not resolve "@elizaos/plugin-image-generation"` issue when trying to import a plug in into the `defaultCharacter.ts` file. It happens with all plugins. 

Example
```
import { imageGenerationPlugin } from "@elizaos/plugin-image-generation";

export const mainCharacter: Character = {
    name: "default",
    plugins: [imageGenerationPlugin],
    clients: [Clients.TWITTER],
    modelProvider: ModelProviderName.OPENAI,
    settings: {
        secrets: {},
        voice: {
```

**Comment by AIFlowML** [comment]
As mentioned is better to use the Json.

**Comment by kaiqiangh** [comment]
> As mentioned is better to use the Json.

Thanks for your suggestion. Any examples for this point? How to specify "plugin" in character.json file?
